yarn-project/builder/src/contract-interface-gen/typescript.ts

```diff
@@ -239,50 +239,51 @@ function generateNotesGetter(input: ContractArtifact) {
   `;
 }
 
-// This is of type AbiType
+// events is of type AbiType
 function generateEvents(events: any[] | undefined) {
   if (events === undefined) {
     return { events: '', eventDefs: '' };
   }
 
-  const eventsStrings = events.map(event => {
-    const eventName = event.path.split('::')[1];
+  const eventsMetadata = events.map(event => {
+    const eventName = event.path.split('::').at(-1);
 
     const eventDefProps = event.fields.map((field: any) => `${field.name}: Fr`);
-    const eventDefs = `
+    const eventDef = `
       export type ${eventName} = {
         ${eventDefProps.join('\n')}
       }
     `;
 
     const fieldNames = event.fields.map((field: any) => `"${field.name}"`);
-    const eventsType = `${eventName}: {decode: (payload: L1EventPayload | undefined) => ${eventName} | undefined }`;
+    const eventType = `${eventName}: {decode: (payload: L1EventPayload | undefined) => ${eventName} | undefined, functionSelector: FunctionSelector, fieldNames: string[] }`;
```

Comment:
@sklppy88 There is an [EventSelector class](https://github.com/AztecProtocol/aztec-packages/blob/06-17-fix_false_decryption_fix/yarn-project/foundation/src/abi/event_selector.ts#L0-L1) and it should be used here instead of FunctionSelector. Could you please do the changes?